File: app/src/main/java/com/rdd/finy/helpers/AppModule.java
package com.rdd.finy.helpers;

import android.app.Application;
import dagger.Module;
import dagger.Provides;

import javax.inject.Singleton;

@Module
public class AppModule {

    private Application mApplication;

    public AppModule(Application application) {
        mApplication = application;
    }

    @Provides
    @Singleton
    Application providesApplication() {
        return mApplication;
    }

}

File: app/src/main/java/com/rdd/finy/helpers/AppModule.kt
package com.rdd.finy.helpers

import android.app.Application
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AppModule(var application: Application) {

    @Provides
    @Singleton
    fun providesApplication():Application{
        return application
    }
}